src/NServiceBus.Core/Routing/UnicastSendRouter.cs

```diff
@@ -49,7 +48,7 @@ UnicastRoute SelectRoute(State state, IOutgoingSendContext context)
                 case RouteOption.RouteToThisInstance:
                     return RouteToThisInstance();
                 case RouteOption.RouteToAnyInstanceOfThisEndpoint:
-                    return UnicastRoute.CreateFromEndpointName(endpointName);
+                    return UnicastRoute.CreateFromEndpointName(receiveQueueName);
```

Comment:
Since receiveQueueName will be null for a send-only endpoint now instead of set to the endpoint name, I wonder if we should add a more specific exception for this case instead of the null guard from `CreateFromEndpointName`.